fix: patch ServiceAccount from a Job instead of a hook

Roming22 opened the pull request:
- Move logic to the script dir
- Improve script
- Add Job
- Add ServiceAccounts

cf [RHTAP-4306](https://issues.redhat.com//browse/RHTAP-4306)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Adds a Helm hook Pod to run post-install/post-upgrade patching of ServiceAccounts across CI, development, stage, and prod namespaces.
  * Deploys per-namespace ServiceAccounts, Roles, and RoleBindings to enable scoped patching.

* **Refactor**
  * Centralizes patching logic into a shared, reusable script and template include for consistent behavior.

* **Chores**
  * Removes legacy post-deploy hook script and standardizes script delivery inside hook pods.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Roming22 commented on installer/charts/tssc-app-namespaces/templates/serviceaccounts/patch-serviceaccounts.yaml:
```diff
@@ -0,0 +1,49 @@
+spec:
+  serviceAccountName: {{ $name }}
+  restartPolicy: Never
+  initContainers:
+    #
+    # Copying the scripts that will be used on the subsequent containers, the
+    # scripts are shared via the "/scripts" volume.
+    #
+    {{- include "common.copyScripts" $context | nindent 8 }}
+  containers:
```
It does, lines 20-22.

dperaza4dustbit reviewed (commented):
Take care of end of line warnings. Looks good otherwise

Roming22 committed:
fix: patch ServiceAccount from a Job instead of a hook

- Move logic to the script dir
- Improve script
- Add Job
- Add ServiceAccounts

cf RHTAP-4306

dperaza4dustbit reviewed (approved):
/lgtm
